### PR TITLE
changes for date issue

### PR DESCRIPTION
**Pull request by Kalyani790**
changes for date issue

### CODE REVIEW

File: app/services/Utils.ts [hunk 1]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
     return dateView;
 }
 export const formatStringDate = (dateData: any): any => {
-    const day = new Intl.DateTimeFormat(luxonDefaultLocale, { day: '2-digit' }).format(new Date(dateData));
-    const month = new Intl.DateTimeFormat(luxonDefaultLocale, { month: '2-digit' }).format(new Date(dateData));
-    const year = new Intl.DateTimeFormat(luxonDefaultLocale, { year: 'numeric' }).format(new Date(dateData));
-    const dateView = day + "." + month + "." + year;
+    dateData=DateTime.fromJSDate(new Date(dateData));
+    console.log(dateData,"..before1");
+    if (typeof dateData == "number") {
```
Comment (Akhror) [inline app/services/Utils.ts:233]: This will never be true won't be?

File: app/services/Utils.ts [hunk 2]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
+        dateData = DateTime.fromMillis(dateData)
+    }
+    console.log(DateTime.fromISO(dateData),".DateTime.fromISO(dateData)")
```
Comment (Akhror) [inline app/services/Utils.ts:236]: I would remove this log since it shows almost the same value as at line 242.

File: app/services/Utils.ts [hunk 3]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
+  
+    //dateData=DateTime.fromJSDate(new Date(dateData));
```
Comment (Akhror) [inline app/services/Utils.ts:238]: We can remove this comment.

File: app/services/Utils.ts [hunk 4]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
+    // const day = new Intl.DateTimeFormat(luxonDefaultLocale, { day: '2-digit' }).format(new Date(dateData));
+    // const month = new Intl.DateTimeFormat(luxonDefaultLocale, { month: '2-digit' }).format(new Date(dateData));
+    // const year = new Intl.DateTimeFormat(luxonDefaultLocale, { year: 'numeric' }).format(new Date(dateData));
+    const dateView = getTwoDigits(day) + "." + getTwoDigits(month) + "." + year;
     return dateView;
 }
-const getTwoDigits = (number: any): string => {
-    return (number < 10 ? '0' : '') + number;
-}
+
```
Comment (Akhror) [inline app/services/Utils.ts:249]: Extra line

File: app/services/Utils.ts [hunk 5]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
 }
+const getTwoDigits = (number: any): string => {
+    return (number < 10 ? '0' : '') + number;
+}
 export const formatDate = (dateData: any): any => {
-    const day = new Intl.DateTimeFormat(luxonDefaultLocale, { day: '2-digit' }).format(new Date(dateData));
-    const month = new Intl.DateTimeFormat(luxonDefaultLocale, { month: '2-digit' }).format(new Date(dateData));
-    const year = new Intl.DateTimeFormat(luxonDefaultLocale, { year: 'numeric' }).format(new Date(dateData));
-    const dateView = day + "." + month + "." + year;
+    dateData=DateTime.fromJSDate(new Date(dateData));
```
Comment (Akhror) [inline app/services/Utils.ts:213]: Let's format the code throughout the change.

File: app/services/Utils.ts [hunk 6]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
         return html;
     }
 }
+const getTwoDigits = (number: any): string => {
+    return (number < 10 ? '0' : '') + number;
+}
```
Comment (Akhror) [inline app/services/Utils.ts:211]: Does number variable always get positive values?

File: app/services/Utils.ts [hunk 7]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
+    if (typeof dateData == "number") {
+        dateData = DateTime.fromMillis(dateData)
+    }
+    console.log(DateTime.fromISO(dateData),".DateTime.fromISO(dateData)")
+  
+    //dateData=DateTime.fromJSDate(new Date(dateData));
+    const day = DateTime.fromISO(dateData).day;
+    const month = DateTime.fromISO(dateData).month;
+    const year = DateTime.fromISO(dateData).year;
+    console.log(day+"."+month+"."+year,"..after1")
```
Comment (Akhror) [inline app/services/Utils.ts:242]: ; is missing at the end

File: app/services/Utils.ts [hunk 8]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
+    dateData=DateTime.fromJSDate(new Date(dateData));
+    console.log(dateData,"..before")
+    if (typeof dateData == "number") {
+        dateData = DateTime.fromMillis(dateData)
+    }
+    console.log(DateTime.fromISO(dateData),".DateTime.fromISO(dateData)")
+    //
+    const day = DateTime.fromISO(dateData).day;
+    const month = DateTime.fromISO(dateData).month;
+    const year = DateTime.fromISO(dateData).year;
```
Comment (Akhror) [inline app/services/Utils.ts:222]: These variables are not needed since we are not using them.

File: app/services/Utils.ts (hunk 2 above)
Comment (Kalyani790) [reply app/services/Utils.ts:236]: ok

File: app/services/Utils.ts [hunk 9]
```diff
@@ -206,23 +206,47 @@ export const addSpaceToHtml = (htmlInput: any): any => {
...
-    const month = new Intl.DateTimeFormat(luxonDefaultLocale, { month: '2-digit' }).format(new Date(dateData));
-    const year = new Intl.DateTimeFormat(luxonDefaultLocale, { year: 'numeric' }).format(new Date(dateData));
-    const dateView = day + "." + month + "." + year;
+    dateData=DateTime.fromJSDate(new Date(dateData));
+    console.log(dateData,"..before")
+    if (typeof dateData == "number") {
+        dateData = DateTime.fromMillis(dateData)
+    }
+    console.log(DateTime.fromISO(dateData),".DateTime.fromISO(dateData)")
+    //
```
[the remark this answers is not in the document]
Comment (Kalyani790) [reply app/services/Utils.ts:219]: ok

File: app/services/Utils.ts (hunk 6 above)
Comment (Kalyani790) [reply app/services/Utils.ts:211]: yes

File: app/services/Utils.ts (hunk 3 above)
Comment (Kalyani790) [reply app/services/Utils.ts:237]: done

File: app/services/Utils.ts (hunk 4 above)
Comment (Kalyani790) [reply app/services/Utils.ts:248]: autoformatting done

File: app/services/Utils.ts (hunk 7 above)
Comment (Kalyani790) [reply app/services/Utils.ts]: console commented

File: app/services/Utils.ts (hunk 8 above)
Comment (Kalyani790) [reply app/services/Utils.ts]: commented

File: app/services/Utils.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (Kalyani790) [reply app/services/Utils.ts]: done

File: app/services/Utils.ts (hunk 2 above)
Comment (Kalyani790) [reply app/services/Utils.ts]: removed

File: app/services/Utils.ts (hunk 1 above)
Comment (Akhror) [reply app/services/Utils.ts:232]: But the DateTime.fromJSDate method always returns a DateTime object regardless of the type of its input argument does not it?

**Pull request merged**